rotel_python_processor_sdk/src/py/mod.rs

```diff
@@ -184,6 +184,18 @@ impl ArrayValue {
             )),
         }
     }
+    fn __setitem__(&self, index: usize, value: &AnyValue) -> PyResult<()> {
+        let mut inner = self.0.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        if index > inner.len() - 1 {
```

Comment:
Likely fine either way, but I feel this avoids any potential underflow problem if len were zero:

```suggestion
        if index >= inner.len() {
```